my-app/src/AddBox/AddBox.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { Component } from 'react';
+import Autosuggest from 'react-autosuggest';
+
+//import isMobile from 'ismobilejs';
+
+//const focusInputOnSuggestionClick = !isMobile.any;
+const uniqLabelFast = a => {
+  /**
+   * Quickly makes a list tags without repeating labels
+   * @param a {[Tag]} list of tags
+   */
+
+    var seen = {};
+    var out = [];
+    var len = a.length;
+    var j = 0;
+    for(var i = 0; i < len; i++) {
+         var item = a[i].label;
+         if(seen[item] !== 1) {
+               seen[item] = 1;
+               out[j++] = a[i];
+         }
+    }
+    return out;
+}
+
+const escapeRegexCharacters = str => str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+
+const getSuggestionValue = (object) => {
+  if (object.hasOwnProperty('label')) {
+    return object.label;
+  } else if (object.hasOwnProperty('name')) {
+    return object.name;
+  }
+}
+
+const getSuggestions = (value, options) => {
+  const escapedValue = escapeRegexCharacters(value.trim());
+  if (escapedValue === '') {
+    return [];
+  }
+  const regex = new RegExp('^' + escapedValue, 'i');
+  return options.filter(tag_or_person => regex.test(getSuggestionValue(tag_or_person)));
+};
+
+
+const renderSuggestion = (suggestion, {query}) => {
+  var img = 'none';
+  if (suggestion.hasOwnProperty('label')) {
```

Comment:
I think it's time to introduce a `class Person` and a `class Tag`, so we can test things like 
```
var a = new Person('Ben', 'ben@ben.com')
var b = new Tag('sweeeeeet')
if (a instanceof Person) { 
  // yes
}
if (a instanceof Tag) {
  // no
}
```

Reply:
Or in another possible style, we could define
```
class Person {
  constructor (name) {
    this.name = name
  } 
  toString () {
    return this.name
  }
}
```
and something similar for a Tag (which also has a toString() method that can be called by renderSuggestion()